### DISCUSSION TITLE
Migrate LB3 datasources to LB4 format

### DISCUSSION THREAD

**Discussion by condratbogdan** [loopbackio/loopback-next#7157] [LB3 Migration]
We have entries like this in our Core service datasources.json file, that represent endpoints in another service named Accounts:
```
"accounts": {
        "name": "accounts",
        "connector": "swagger",
        "spec": "http://192.168.100.103:3001/explorer/swagger.json"
    }
```
We could then use the endpoints from the service inside our loopback app like this, Accounts.Accounts_findById({...}).
How can i get the same effect using LB4 components? (both creation of datasource and method invokation)

**Comment by jannyHou** [comment]
Hi @condratbogdan when you say "How can i get the same effect using LB4 components", do you mean turn the app with accounts services into a LB4 component? Or you want to invoke the account service in a LB4 component?

**Comment by condratbogdan** [comment]
> Hi @condratbogdan when you say "How can i get the same effect using LB4 components", do you mean turn the app with accounts services into a LB4 component? Or you want to invoke the account service in a LB4 component?

I want to invoke the account service in a LB4 component.

**Comment by bajtos** [comment]
In LoopBack 4, we use the concept of a Service (or a Service Proxy) to connect the application to 3rd party web services. Please take a look at our documentation page [Calling REST APIs](https://loopback.io/doc/en/lb4/Calling-rest-apis.html), it shows how to define a datasource and invoke remote service methods. I _think_ our OpenAPI connector supports Swagger too (Swagger is v2.0, OpenAPI is v3.0).
